### PR TITLE
Remove cucumber, dco, knife-spork

### PR DESCRIPTION
**Pull request by marcparadise**
These tools are not used by Workstation, and do not fit into any recommended workflow. 

Signed-off-by: Marc A. Paradise <marc.paradise@gmail.com>

### CODE REVIEW

File: components/gems/Gemfile [hunk 1]
```diff
@@ -81,7 +81,6 @@ group(:omnibus_package) do
   gem "net-ssh", ">= 4.2.0"
   gem "test-kitchen", ">= 2.0"
   gem "listen"
-  gem "dco"   # TODO - dksplit - Planned for removal
```
Comment (robbkidd) [inline components/gems/Gemfile]: I know that many (most? all?) of the Workstation project maintainers probably have git configured to do the commit sign-offs. I'm not so sure about the rest of the community. Did I miss some conversations about its utility?

Comment (marcparadise) [reply components/gems/Gemfile]: The dco gem was included undocumented to ease the transition to the DCO process; the initial blog post details the instructions for configuring git and signing commits, but does not refer anyone to the dco tool to manage this. 

Given that the transition to DCO has gone smoothly, and the tool itself is not publicized in DK, there seems little value in continuing to maintain it in Workstation.

Comment (robbkidd) [reply components/gems/Gemfile]: Fair enough.

**Comment by marcparadise** [comment]
This bump to chef-cli should see verification passing.

**Pull request merged**